### PR TITLE
Jgordon missing tracking number

### PR DESCRIPTION
**Pull request by jathmel**

### CODE REVIEW

File: src/Controllers/ShippingController.php [hunk 1]
```diff
@@ -55,6 +55,12 @@ public function registerShipments(Request $request, array $orderIds): array {
             'method' => __METHOD__
             ]
         );
+    $this->loggerContract->debug(
+            TranslationHelper::getLoggerKey('debugRegisterShipmentForOrders'), [
```
Comment (jhoule86) [inline src/Controllers/ShippingController.php:59]: use a constant for logger code, or remove log if no longer needed.

File: src/Controllers/ShippingController.php [hunk 2]
```diff
@@ -55,6 +55,12 @@ public function registerShipments(Request $request, array $orderIds): array {
...
+            'additionalInfo' => ['orderIds' => $orderIds, 'request' => $request],
```
Comment (jhoule86) [inline src/Controllers/ShippingController.php:60]: Have you verified that the `request` object is visible in the logger UI?

File: src/Core/Api/Services/FetchOrderService.php [hunk 3]
```diff
@@ -34,6 +34,14 @@ public function fetch(int $circle): array
     $query = $this->getQuery($circle);
 
     $response = $this->query($query);
+    // $loggerContract->debug('Test',
+    $loggerContract->debug(TranslationHelper::getLoggerKey('test'),
```
Comment (jhoule86) [inline src/Core/Api/Services/FetchOrderService.php:38]: Looks like leftover debugging code?

File: src/Cron/OrderAcceptCron.php [hunk 4]
```diff
@@ -50,6 +50,7 @@ public function handle() {
     $loggerContract->debug(TranslationHelper::getLoggerKey('cronStartedMessage'), ['method' => __METHOD__]);
     try{
       $this->orderService->accept($externalLogs, 1);
+      // $loggerContract->debug(TranslationHelper::getLoggerKey('test'), ['additionalInfo'=> $this->orderService->accept($externalLogs, 1),'method' => __METHOD__]);
```
Comment (jhoule86) [inline src/Cron/OrderAcceptCron.php]: remove

File: src/Cron/OrderImportCron.php [hunk 5]
```diff
@@ -53,6 +53,9 @@ public function handle() {
       $this->loggerContract->debug(TranslationHelper::getLoggerKey('cronStartedMessage'), ['method' => __METHOD__]);
       $this->orderService->process($externalLogs, 1);
     }
+    catch(\Exception $e) {
+      $this->loggerContract->error(TranslationHelper::getLoggerKey('test'), ['additionalInfo'=> $this->orderService->process($externalLogs, 1) ,'message' => $e->message, 'method' => __METHOD__]);
```
Comment (jhoule86) [inline src/Cron/OrderImportCron.php]: change to a different logger key (declared as a constant). We don't want to eat this Exception, but `test` can't be the desired logger key.

File: src/Providers/WayfairServiceProvider.php [hunk 6]
```diff
@@ -76,7 +76,7 @@ public function boot(
     EventProceduresService $eventProceduresService
   ) {
     // register crons
-    $cronContainer->add(CronContainer::EVERY_FIFTEEN_MINUTES, OrderImportCron::class);
+    $cronContainer->add(CronContainer::EVERY_FIVE_MINUTES, OrderImportCron::class);
```
Comment (jhoule86) [inline src/Providers/WayfairServiceProvider.php:79]: Revert - we have no plans to change the period of this cron job.

File: src/Services/OrderPropertyService.php [hunk 7]
```diff
@@ -97,6 +97,17 @@ public function getCheckedPoNumber(int $orderId): string {
...
+        TranslationHelper::getLoggerKey(self::LOG_KEY_WAREHOUSE_ID_NOT_FOUND),
+        [
+          'additionalInfo' => [
+            'orderId' => $orderId,
+            'typeId' => OrderPropertyType::WAREHOUSE,
+          'order properties' => $orderProperties
+          ],
+          'method' => __METHOD__
+        ]
+      );
```
Comment (jhoule86) [inline src/Services/OrderPropertyService.php:110]: What is going on here? This is an unconditional log of `LOG_KEY_WAREHOUSE_ID_NOT_FOUND` **outside** of the conditional that checks for warehouse info.

File: src/Services/ShipmentNotificationService.php [hunk 8]
```diff
@@ -331,20 +332,19 @@ public function prepareRequestDto(Order $order): RequestDTO
...
-              'additionalInfo' => [
-                'orderId' => $orderId,
-                'poNumber' => $poNumber
-              ],
-              'method' => __METHOD__,
-              'referenceType' => '$orderId',
-              'referenceValue' => $orderId
-            ]
-          );
+      $plentymarketsShippingInformation = json_encode ($this->shippingInformationRepositoryContract->getShippingInformationByOrderId($orderId));
```
Comment (jhoule86) [inline src/Services/ShipmentNotificationService.php:336]: I think the `plentymarketsShippingInformation` object is being used later, right? So is it okay to change it to a string by json encoding it?

File: src/Services/ShipmentNotificationService.php [hunk 9]
```diff
@@ -368,7 +369,8 @@ public function prepareRequestDto(Order $order): RequestDTO
       $products = $purchaseOrderInfo['products'];
 
       //Decide how to get tracking and package information.
-      if ($this->shipmentProviderService->isShippingWithWayfair() || empty($plentymarketsShippingInformation->shippingServiceProvider)) {
+      // if ($this->shipmentProviderService->isShippingWithWayfair()) {
+      if ($this->shipmentProviderService->isShippingWithWayfair()){ //|| empty($plentymarketsShippingInformation->shippingServiceProvider)) {
```
Comment (jhoule86) [inline src/Services/ShipmentNotificationService.php:374]: Where/how do we handle the situation of no `shippingServiceProvider` after these changes?

File: src/Services/ShipmentNotificationService.php [hunk 10]
```diff
@@ -356,7 +356,8 @@ public function prepareRequestDto(Order $order): RequestDTO
           'additionalInfo' => [
             'order' => $order,
             'po' => $purchaseOrderInfo,
-            'shippingInformation' => $plentymarketsShippingInformation
+            'shippingInformation' => $plentymarketsShippingInformation,
+            'ShippingOnWayfair' => $this->shipmentProviderService->isShippingWithWayfair()
```
Comment (jhoule86) [inline src/Services/ShipmentNotificationService.php:361]: identifier in log should be camelCase

File: src/Services/ShipmentNotificationService.php [hunk 11]
```diff
@@ -331,20 +332,19 @@ public function prepareRequestDto(Order $order): RequestDTO
...
-                'orderId' => $orderId,
-                'poNumber' => $poNumber
-              ],
-              'method' => __METHOD__,
-              'referenceType' => '$orderId',
-              'referenceValue' => $orderId
-            ]
-          );
+      $plentymarketsShippingInformation = json_encode ($this->shippingInformationRepositoryContract->getShippingInformationByOrderId($orderId));
+      if (!isset($plentymarketsShippingInformation) || empty($plentymarketsShippingInformation) || empty($plentymarketsShippingInformation->shippingServiceProvider)) {
```
Comment (jhoule86) [inline src/Services/ShipmentNotificationService.php]: how is `$plentymarketsShippingInformation->shippingServiceProvider` not throwing errors if you made it a `string` on the line above this one?

File: src/Services/ShipmentNotificationService.php [hunk 12]
```diff
@@ -495,6 +500,18 @@ public function prepareRequestDto(Order $order): RequestDTO
         $orderTrackingNumbers = $this->orderRepositoryContract->getPackageNumbers($orderId);
         $requestDto->setPackageCount(count($orderShippingPackages) > 0 ? count($orderShippingPackages) : 1);
 
+        $this->loggerContract->debug(
+          TranslationHelper::getLoggerKey(self::LOG_KEY_SHIPPING_ON_OWN_ACCOUNT),
```
Comment (jhoule86) [inline src/Services/ShipmentNotificationService.php:517]: It looks like this log is now going to appear twice, with slightly different info, at different levels?

File: src/Services/ShipmentRegisterService.php [hunk 13]
```diff
@@ -268,6 +269,17 @@ public function register(array $orderIds): array
...
+            TranslationHelper::getLoggerKey(self::LOG_KEY_DEBUG_REGISTER_INFO),
+            [
+              'additionalInfo' => [
+                'orderID' => $orderId,
+                'message' => 'ShipmentRegister service line 294',
+                'shippingInformation' => $shippingInformation
+              ],
+              'method' => __METHOD__
+            ]
+          );
```
Comment (jhoule86) [inline src/Services/ShipmentRegisterService.php:282]: Looks like overly-specific leftover debug code (with line numbers).

File: src/Controllers/ShippingController.php (hunk 2 above)
Comment (jathmel) [reply src/Controllers/ShippingController.php:60]: This was to be removed, I am unable to see the `request object` in the logger UI

File: src/Services/ShipmentRegisterService.php (hunk 13 above)
Comment (jathmel) [reply src/Services/ShipmentRegisterService.php:282]: This was to help with pin-pointing the specific log I will generalize

File: src/Cron/OrderImportCron.php (hunk 5 above)
Comment (jathmel) [reply src/Cron/OrderImportCron.php]: What do you mean by eat?

Comment (jhoule86) [reply src/Cron/OrderImportCron.php]: Meaning we want to log the Exception's details, but `test` is not going to be the right key here.

File: src/Services/ShipmentRegisterService.php [hunk 14]
```diff
@@ -42,6 +42,7 @@ class ShipmentRegisterService
   const LOG_KEY_SHIPPING_LABEL_RETRIEVAL_FAILED = 'shippingLabelRetrievalFailed';
   const LOG_KEY_NO_SHIPPING_INFO_FOR_UNREGISTER = 'noShippingInformationForUnregister';
   const LOG_KEY_WAREHOUSE_MISSING_FOR_ORDER = 'warehouseMissingForOrder';
+  const LOG_KEY_DEBUG_REGISTER_INFO = 'registerDebuger';
```
Comment (jhoule86) [inline src/Services/ShipmentRegisterService.php:45]: I don't think `registerDebuger` is in the `logger.properties` files.

File: src/Cron/OrderImportCron.php (hunk 5 above)
Comment (jathmel) [reply src/Cron/OrderImportCron.php]: Ok thank you for the clarification

**Pull request closed without merging**